src/item_group.cpp

```diff
@@ -415,14 +415,10 @@ void Item_modifier::modify( item &new_item, const std::string &context ) const
 
     // create container here from modifier or from default to get max charges later
     item cont;
-    if( container != nullptr ) {
+    if( container != nullptr && cont.is_null() ) {
```

Comment:
```suggestion
    if( container != nullptr ) {
```

`cont` is always null in these checks, so it can also go.